models/demos/t3000/falcon40b/tests/test_perf_falcon.py

```diff
@@ -408,3 +408,75 @@ def test_perf_bare_metal(
         expected_inference_time,
         warmup_iterations=10,
     )
+
+
+@skip_for_grayskull("Requires eth connected devices to run")
+@pytest.mark.parametrize("num_devices", (8,), ids=["8chips"])
+@pytest.mark.parametrize(
+    "llm_mode, batch, seq_len, kv_cache_len, expected_compile_time, expected_inference_time, num_layers, model_config_str",
+    (
+        ("prefill", 1, 128, 0, 60, 0.39 + 0.04, 1, "BFLOAT8_B-DRAM"),
+        ("prefill", 1, 2048, 0, 60, 0.94 + 0.1, 1, "BFLOAT8_B-DRAM"),
+    ),
+    ids=[
+        "prefill_seq128_bfp8_layers1",
+        "prefill_seq2048_bfp8_layers1",
+    ],
+)
+@pytest.mark.parametrize(
+    "model_version",
+    ("tiiuae/falcon-40b-instruct",),
+    ids=["falcon_40b"],
+)
+def test_device_perf_bare_metal(
```

Comment:
what's the difference between this test and test_perf_bare_metal - just parametrizations? 
why do we need warmup loops if we measure device perf?

Reply:
Yeah, only difference is parameterization - for device we want 1 layer and for end to end we want 60 layers. The device tests doesn't have the `@pytest.mark.model_perf_t3000` so it's not included in the perf pipeline.

Reply:
I saw the 1 layer configs were also in the `test_perf_bare_metal` test, removed them now.

Reply:
warmup loops: I think this is required for both device and e2e perf measurements to get the cpu in performance mode? or is that only relevant for e2e perf?

Reply:
Correct, cpu needs to be warmed up, but with device perf we are interested in op time on the device we get from Tracy, so it shouldn't be relevant?